test/package/install/pvt_pkg_install_test.go

```diff
@@ -0,0 +1,81 @@
+package private_package
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+
+	// corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	pp "github.com/crossplane/test/apis/package"
+)
+
+func TestPackageInstall(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		body   func(privatePackage string, InstallPackage pp.InstallPackage) error
+	}{
+		"InstallPrivatePackage": {
+			reason: "Should be able to successfully install a private package from registries supported by Crossplane.",
+			body: func(privatePackage string, InstallPackage pp.InstallPackage) error {
+				sl := strings.SplitAfter(privatePackage, "/")
+				InstalledPackageName := sl[len(sl)-1]
```

Comment:
@parul5sahoo it may be easier to use `go-containerregistry` to reliably parse package name from image reference. See example of `name.ParseReference` [here](https://github.com/crossplane/crossplane/blob/b7ce021e32305e60ed920a48883c03b332e556e7/internal/controller/pkg/revision/dependency.go#L110).